services/nuget/nuget-v3-service-family.js

```diff
@@ -184,16 +185,19 @@ function createServiceFamily({
         feed,
       })
       const { versions } = await fetch(this, { baseUrl, packageName })
-
       let latest = versions.slice(-1).pop()
       const includePrereleases = which === 'vpre'
       if (!includePrereleases) {
-        const filtered = versions.filter(item => !item.version.includes('-'))
+        const filtered = versions.filter(item => {
+          if (semver.valid(item.version)) {
+            return !semver.prerelease(item.version)
+          }
+          return !item.version.includes('-')
```

Comment:
Not clear how likely it is for us to hit this line as I'm not really clear what the constraints are on a NuGet version string, but how about


```js
return !item.version.split('+')[0].includes('-')
```

here? (i.e: we only say its a pre-release if the `-` character appears before a `+` character)

Reply:
> I'm not really clear what the constraints are on a NuGet version string

Me either, but my sense is that there aren't really any constraints, or at least that the version formats vary wildly.  I wanted to handle the semver specific use case without risking rocking the boat 😄 

Are you suggesting replacing the entire check with the `return !item.version.split('+')[0].includes('-')`, or just replacing line 195?

Reply:
If you were curious for details on package versioning for NuGet, there is some documentation here: https://docs.microsoft.com/en-au/nuget/concepts/package-versioning

The tl;dr, it is close to (if not exactly) semver v1 for some packages and semver v2 otherwise.

Reply:
thanks @Turnerj! Do you know if that is enforced by all the various registries that support nuget packages though? Or is it just convention/just enforced by some, like nuget.org?

Reply:
I don't believe it is enforced by all registries. NuGet is definitely the biggest registry though (its the default in Visual Studio), probably followed by MyGet (though likely way behind NuGet) before then counting the more ad-hoc things like CI's providing NuGet feeds of builds.

I can't find any specific details about what MyGet does with their NuGet feed support but from what I gleam from their docs, I think they likely follow closely what NuGet do. Regarding other ad-hoc ones, I doubt there is any specific constraint handling for package versions.

Reply:
I suppose it all depends on whether the `-` before `+` precedence applies in non-semver compatible versions. 

Admittedly a heavily contrived example, but let's say there's a nuget package that contains one or more non-semver compliant versions like:
`1b+abc-123`

We'd currently classify that as a pre-release version, and I don't know if/how non-disruptively we can change how we treat that to say it's not a pre-release because the SemVer spec says that everything after the `+` is build metadata which can include a `-` character without the version being pre-release.

TBH I'm not sure what the correct answer is, and I don't feel strongly one or the other. If we want to say that when determining whether a nuget package version is pre-release we won't include any `-` characters that occur after a `+` character then I'm 👍 with that, and I'll add some inline comments with links to this thread. 

And if we do, then I think we can get by with just using `!item.version.split('+')[0].includes('-')` since we're going to apply that rule whether the version is semver compliant or not (`semver.prerelease` is functionally doing the same thing anyway).

Reply:
Yeah if version can be an arbitrary string, its probably meaningless trying to decide if its a pre-release or not in the fallback case where its not a semver.

Reply:
That sounds fair at least - if you want to handle pre-releases, use semver for your versioning.

Reply:
I agree with @chris48s and @Turnerj. However, to keep the badges consistent with how they are currently displayed for users who do not use semantic versioning, I suggest we leave line 195 as it is above, i.e. no changes to this part of the PR. If someone opens an issue for a non semantic versioning case, we can discuss it then and see if we can refine our handling. Does this sound good?